components/TxForm.tsx

```diff
@@ -29,6 +52,108 @@ export const TxForm = () => {
+    args: [
+      getRailgunSmartWalletContractForNetwork(NetworkName.EthereumGoerli)
+        .address,
+      constants.MaxUint256,
+    ],
+  });
+  const { writeAsync: doErc20Approval } = useContractWrite(config);
+
+  const doSubmit: React.FormEventHandler = async (e) => {
+    // Formatted token amounts and recipients.
+    const erc20AmountRecipients: RailgunERC20AmountRecipient[] = [
+      {
+        tokenAddress: "0xb4fbf271143f4fbf7b91a5ded31805e42b2208d6", // goerli weth
+        amountString: "0x10", // hexadecimal amount
+        recipientAddress:
+          "0zk1qyn0qa5rgk7z2l8wyncpynmydgj7ucrrcczhl8k27q2rw5ldvv2qrrv7j6fe3z53ll5j4fjs9j5cmq7mxsaulah7ykk6jwqna3nwvxudp5w6fwyg8cgwkwwv3g4", // RAILGUN address
+      },
+    ];
+
+    // The shieldPrivateKey enables the sender to decrypt
+    // the receiver's address in the future.
+    const shieldPrivateKey = keccak256(
+      await (signer as Signer).signMessage(
+        getShieldPrivateKeySignatureMessage()
+      )
+    );
+
+    // if (!doErc20Approval) throw "not prepared";
+    // await doErc20Approval();
+
+    // Public wallet to shield from.
+    const fromWalletAddress = address as `0x{string}`;
+
+    const { gasEstimateString, error: err } = await gasEstimateForShield(
+      NetworkName.EthereumGoerli,
+      shieldPrivateKey,
+      erc20AmountRecipients,
+      [], // nftAmountRecipients
+      fromWalletAddress
+    );
+
+    if (err) {
+      throw err;
+    }
+
+    const { maxFeePerGas, maxPriorityFeePerGas } = await provider.getFeeData();
+    const gasDetailsSerialized: TransactionGasDetailsSerialized = {
+      evmGasType: EVMGasType.Type2, // Depends on the chain (BNB uses type 0)
+      gasEstimateString: gasEstimateString!, // Output from gasEstimateForShield
+      maxFeePerGasString: maxFeePerGas!.toHexString(), // Current gas Max Fee
+      maxPriorityFeePerGasString: maxPriorityFeePerGas!.toHexString(), // Current gas Max Priority Fee
+    };
+
+    const { serializedTransaction, error } = await populateShield(
+      NetworkName.EthereumGoerli,
+      shieldPrivateKey,
+      erc20AmountRecipients,
+      [], // nftAmountRecipients
+      gasDetailsSerialized
+    );
+    if (error) {
+      throw error;
+    }
+    console.log(gasEstimateString);
+    const gasEstimate = BigNumber.from(gasEstimateString);
+    console.log(gasEstimate);
+
+    const { chain } = NETWORK_CONFIG[NetworkName.EthereumGoerli];
+
+    const transactionRequest: ethers.providers.TransactionRequest =
+      deserializeTransaction(
+        serializedTransaction as string,
+        undefined, // nonce (optional)
+        chain.id
+      );
+
+    // Public wallet to shield from.
+    transactionRequest.from = address;
+    const tx = await signer?.sendTransaction(transactionRequest);
+    console.log(tx);
+    await tx?.wait().then(() => {
+      notifyUser({
+        alertType: "success",
+        message: "Token was shielded successfully",
+      });
+    });
```

Comment:
Let's make sure we're handling all scenarios here: https://docs.ethers.org/v5/single-page/#/v5/api/providers/types/-%23-providers-TransactionResponse

If the tx reverts the promise rejects so we should catch that. We should also get the receipt returned by `tx.wait()` and sanity check the status is 1, not zero

Reply:
yeah, down for all that; given that transaction handling will likely be relocated and dealt with in another feature i'm inclined to leave as is here with a TODO, what do you think?

Reply:
TODO works for me!

Reply:
added TODO: https://github.com/ScopeLift/railgun-shielder/pull/34/files#diff-809389dcad3e410c420c6a1e2225d2a9a33a40fa577026c2b17ca4e51f73f36dR134